pkg/intermediate/aggregate.go

```diff
@@ -648,6 +712,132 @@ func (a *AggregationProcess) addFieldsForStatsAggregation(record entities.Record
 	return nil
 }
 
+func (a *AggregationProcess) addFieldsForThroughputCalculation(record entities.Record, fillSrcStats, fillDstStats bool) error {
+	if a.aggregateElements == nil {
+		return nil
+	}
+	antreaFlowEndSecondsElements := a.aggregateElements.AntreaFlowEndSecondsElements
+	antreaThroughputElements := a.aggregateElements.ThroughputElements
+	antreaSourceThroughputElements := a.aggregateElements.SourceThroughputElements
+	antreaDestinationThroughputElements := a.aggregateElements.DestinationThroughputElements
+
+	var timeStart, timeEnd uint32
+	var byteCount, reverseByteCount uint64
+	timeStart, err := getUnsigned32ValueByIeName(record, "flowStartSeconds")
+	if err != nil {
+		return err
+	}
+	timeEnd, err = getUnsigned32ValueByIeName(record, "flowEndSeconds")
+	if err != nil {
+		return err
+	}
+	byteCount, err = getUnsigned64ValueByIeName(record, "octetTotalCount")
+	if err != nil {
+		return err
+	}
+	reverseByteCount, err = getUnsigned64ValueByIeName(record, "reverseOctetTotalCount")
+	if err != nil {
+		return err
+	}
+
+	// Initialize flowEndSecondsFromSourceNode and flowEndSecondsFromDestinationNode.
+	for _, ieName := range antreaFlowEndSecondsElements {
+		ie, err := registry.GetInfoElement(ieName, registry.AntreaEnterpriseID)
+		if err != nil {
+			return err
+		}
+		value := uint32(0)
+		if fillSrcStats && strings.Contains(ieName, "Source") || fillDstStats && strings.Contains(ieName, "Destination") {
+			value = timeEnd
+		}
+		if err = record.AddInfoElement(entities.NewUnsigned32InfoElement(ie, value)); err != nil {
+			return err
+		}
+	}
+
+	// Initialize the throughput elements.
+	var incomingVal, reverseIncomingVal uint64
+	// For the edge case when the record has the same timeEnd and timeStart values,
```

Comment:
Change: Adding an edge case handling, since for some of the conntrack connections, when the start time is zero, both start and end time will be set reset to the current time - [code](https://github.com/antrea-io/antrea/blob/c135609aaa9b63d74842c7cf2f7981d1c44379f4/pkg/agent/flowexporter/connections/conntrack_connections.go#L242)